FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -0,0 +1,428 @@
+
+                                var nextJobEnd = (DateTimeOffset)nextResourceTimestamp + interval;
+                                if (nextJobEnd > _inputData.DataEndTime)
+                                {
+                                    nextJobEnd = _inputData.DataEndTime;
+                                }
+
+                                input = new FhirToDataLakeProcessingJobInputData
+                                {
+                                    JobType = JobType.Processing,
+                                    ProcessingJobSequenceId = _result.CreatedJobCount,
+                                    TriggerSequenceId = _inputData.TriggerSequenceId,
+                                    Since = _inputData.Since,
+                                    DataStartTime = nextResourceTimestamp,
+                                    DataEndTime = nextJobEnd,
+                                };
+                                _result.NextJobTimestamp = nextJobEnd;
+                            }
+
+                            break;
+                        case FilterScope.Group:
+                            if (_result.NextPatientIndex < toBeProcessedPatients.Count)
+                            {
+                                var selectedPatients = toBeProcessedPatients.Skip(_result.NextPatientIndex)
+                                    .Take(NumberOfPatientsPerProcessingJob).ToList();
+                                input = new FhirToDataLakeProcessingJobInputData
+                                {
+                                    JobType = JobType.Processing,
+                                    ProcessingJobSequenceId = _result.CreatedJobCount,
+                                    TriggerSequenceId = _inputData.TriggerSequenceId,
+                                    Since = _inputData.Since,
+                                    DataStartTime = _inputData.DataStartTime,
+                                    DataEndTime = _inputData.DataEndTime,
+                                    ToBeProcessedPatients = selectedPatients,
+                                };
+                                _result.NextPatientIndex += selectedPatients.Count;
+                            }
+
+                            break;
+                        default:
+                            // this case should not happen
+                            throw new ArgumentOutOfRangeException(
+                                $"The filterScope {_filterConfiguration.FilterScope} isn't supported now.");
+                    }
+
+                    if (input == null)
+                    {
+                        break;
+                    }
+
+                    var jobDefinitions = new[] { JsonConvert.SerializeObject(input) };
+                    var jobInfos = await _queueClient.EnqueueAsync(
+                        _jobInfo.QueueType,
+                        jobDefinitions,
+                        _jobInfo.GroupId,
```

Comment:
Do we need to handle conflict here?

Reply:
Actually, we don't need to handle conflict here. If they are the same processing job enqueued, the existing jobInfo will be returned.

Reply:
Per our offline discussion, it is OK here. We have unit tests for re-enqueue with different statuses.